app/reconciliation.go

```diff
@@ -0,0 +1,116 @@
+package app
+
+import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/store"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/bnb-chain/node/common"
+	"github.com/bnb-chain/node/common/types"
+)
+
+const globalAccountNumber = "globalAccountNumber"
+
+// reconBalance will do reconciliation for accounts balances.
+func (app *BinanceChain) reconBalance(ctx sdk.Context) {
+	currentHeight := ctx.BlockHeight()
+
+	accountStore, ok := app.GetCommitMultiStore().GetCommitStore(common.AccountStoreKey).(*store.IavlStore)
+	if !ok {
+		panic("cannot convert account store to ival store")
+	}
+	accPre, accCurrent := app.getAccountChanges(ctx, accountStore)
+	accountStore.ResetDiff()
+
+	tokenStore, ok := app.GetCommitMultiStore().GetCommitStore(common.TokenStoreKey).(*store.IavlStore)
+	if !ok {
+		panic("cannot convert token store to ival store")
+	}
+	tokenPre, tokenCurrent := app.getTokenChanges(ctx, tokenStore)
+	tokenStore.ResetDiff()
+
+	left := tokenPre.Plus(accCurrent)
```

Comment:
I am afraid that it is possible to overflow here.

Reply:
refactor the code for comparing